pkg/controller/argocd/keycloak.go

```diff
@@ -46,8 +47,23 @@ var (
 	controllerRef  bool  = true
 )
 
-// getKeycloakContainerImage will return the container image for Keycloak.
-func getKeycloakContainerImage(img string, ver string) string {
+// getKeycloakContainerImage will return the container image for the Keycloak.
+//
+// There are two possible options for configuring the image, and this is the
+// order of preference.
+//
+// 1. From the Environment, this looks for the `ARGOCD_KEYCLOAK_IMAGE` field and uses
+// that.
+// 2. the default is configured in common.ArgoCDKeycloakVersion and
+// common.ArgoCDKeycloakImageName.
+func getKeycloakContainerImage() string {
```

Comment:
In case of Keycloak, I think we should not be providing an option for `ImageName` in CR. RHSSO is running a very old version of Keycloak, If we provide this option in CR, we are opening it up for Users to provide an Image of their choice which might not work as latest versions of Keycloak has changed Drastically.

Reply:
We do this for all other images.  The age of the version is not a factor to justify going against the pattern.   You can say the same to dex or any other images configurable in CR can open up for users to use a bad/incompatible images.   Making a variant for keycloak implementation means more difficult to maintain the code.

Reply:
acknowledge